azurerm/modules/azurerm-adls/variables.tf

```diff
@@ -0,0 +1,90 @@
+############################################
+# RESOURCE INFORMATION
+############################################
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource Group Name"
+}
+
+variable "resource_group_location" {
+  description = "Location of the resource group"
+  type        = string
+  default     = "uksouth"
+}
+
+############################################
+# STORAGE ACCOUNT SETTINGS
+############################################
+
+variable "account_replication_type" {
+  type        = string
+  description = "The Storage Account replication type. Valid options are LRS, GRS, RAGRS, ZRS, GZRS and RAGZRS."
+  default     = "LRS"
+}
+
+variable "container_access_type" {
+  type        = string
+  description = "value"
+  default     = "The Access Level configured for this Container. Possible values are blob, container or private. Defaults to private."
+}
+
+############################################
+# NAMING
+############################################
+variable "resource_tags" {
+  description = "Map of tags to be applied to all resources created as part of this module"
+  type        = map(string)
+  default     = {}
+}
+
+variable "storage_account_name" {
```

Comment:
Just thinking, should this be `resource_namer` to fit in with how this is done in other modules? This will essentially be the prefix of the storage account name, with the `storage_account_details` name appended on the end.

Reply:
@adurkan-amido Used the resource namer now on the resource level instead of data modulation during the module call: 

```
name                     = substr(replace("${var.resource_namer}${each.value.name}", "-", ""), 0, 24)